src/CPSubsystem.ts

```diff
@@ -119,9 +123,28 @@ export class CPSubsystemImpl implements CPSubsystem {
     private readonly cpProxyManager: CPProxyManager;
     private readonly cpSessionManager: CPSessionManager;
 
-    constructor(client: HazelcastClient) {
-        this.cpProxyManager = new CPProxyManager(client);
-        this.cpSessionManager = new CPSessionManager(client);
+    constructor(
+        clientConfig: ClientConfig,
+        logger: ILogger,
+        clientName: string,
+        invocationService: InvocationService,
+        serializationService: SerializationService,
+        connectionManager: ClientConnectionManager
```

Comment:
I removed cyclic depen between connection manager and invocation service by passing connection manager to invoke() methods. In the last version there are no services that depend on each other on construction. There are two interfaces that only have some client methods in them.

Reply:
That's great. Going to look at the current version.